### ISSUE TITLE
every item in toc is preceded by "{:.}"

### ISSUE THREAD

**Issue by claresudbery** [allejo/jekyll-toc#28]
**Public Repo**

My repo is private, but based on this public repo from Ably: https://github.com/ably/wiki-site

**Description**

I've just started using the jekyll-toc as an include in my Jekyll site, but every item in the TOC is preceded by "{:.}". I'm new to Jekyll and Liquid so I don't understand why this appears or what it means?

**Your Usage**

This is a simple vanilla usage. I downloaded toc.html, placed it in my _includes directory and added the following line to my _layouts/default.html:

```html
{% include toc.html html=content %}
```

I tried to change its behaviour by adding sanitize=true, but it didn't make any difference.

Here is a screenshot:

![toc](https://user-images.githubusercontent.com/8990232/72148925-e698ae00-3399-11ea-887c-c28e2e144f8d.PNG)

**Comment by claresudbery** [comment]
I figured out a solution. I don't know why these characters were in toc.html, but when I removed them it fixed the problem.

I changed it from this:

```html
{% unless include.item_class == blank %}
            {% capture listItemClass %}{:.{{ include.item_class | replace: '%level%', headerLevel }}}{% endcapture %}
        {% endunless %}
```

... to this:

```html
{% unless include.item_class == blank %}
            {% capture listItemClass %}{{ include.item_class | replace: '%level%', headerLevel }}{% endcapture %}
        {% endunless %}
```

**Comment by allejo** [comment]
The `{:.}` that shows up is the markdown directive to add a class to an element. Can you confirm a few things?

- Your headings are in order (see #6)
- You're using kramdown in your site (which is the Jekyll default)

**Comment by claresudbery** [comment]
Yes I'm using kramdown. I have h2s with no h1s, so that might explain it.

**Comment by allejo** [comment]
> Yes I'm using kramdown. I have h2s with no h1s, so that might explain it.

The lack of `h1`s shouldn't matter. Any chance you could create a dummy repo/file to replicate this issue? Because just going off of the unit tests in this repo, this *should* be working.

**Comment by claresudbery** [comment]
Hi 

Yes, I can do that.

Either later this evening or tomorrow.

Cheers 
Clare.

**Comment by allejo** [comment]
This was being caused by using redcarpet as the markdown engine instead of kramdown, see [`ably/wiki-site:_config.yml`](https://github.com/ably/wiki-site/blob/8776db6513a7132a9101fb0544d41815d93d2e8e/_config.yml#L8-L13).

However, this did point out another logic issue in the Liquid where it was generating code it shouldn't, which was fixed in #29.
